### DISCUSSION TITLE
Link-style button in text area

### DISCUSSION THREAD

**Discussion by pguptaDS** [twilio-labs/paste#1738] [Q&A] [answered]
Hello! I am migrating some old designs to paste and was wondering if its possible to replicate this behavior using paste components? Essentially, the functionality we want is being able to collapse/expand text within a read-only text area.

Here's what the old designs look like: 
Collapsed:
![image](https://user-images.githubusercontent.com/64559618/128119456-78f646b4-d53c-49ae-a571-4be2e4941786.png)

Expanded:
![image](https://user-images.githubusercontent.com/64559618/128119481-708da3b3-17a1-4b87-9535-069c67a613a0.png)

**Comment by richbachman** [comment]
Hi @pguptaDS I'm checking in with the team on this one. We'll be back in touch asap.

**Reply by pguptaDS** [reply]
Hello @richbachman! Just wanted to check in on this. Please let me know if you have any questions for me.

**Comment by richbachman** [comment, marked as answer]
Hi @pguptaDS 

We've been discussing this as a team, and I mocked up a quick prototype to show how this might look using Paste components. This is purely presentational so none of the buttons actually work.

https://codesandbox.io/s/copy-cert-example-r6px2?file=/src/index.tsx

There's a couple things we'd need to work-through:

1. How important is the expand / collapse functionality? If its necessary, we'll have to add some for consumers to control the height of the text area. Right now the height is determined by the content, along with a max-height. I'm currently discussing options with team.
2. The copy button functionality is a work in progress as well. We should have a copy to clipboard library in the next Paste release.

We'll be back in touch as soon as we have more info. Thanks for your patience.

**Reply by pguptaDS** [reply]
Got it. Thanks for showing how it could work. 
Quick clarification - the expand/collapse button would ideally be within the text area so it doesn't disrupt the content that is below the text area such as other buttons etc.
Looking forward to what y'all come up with!

**Reply by richbachman** [reply]
Another idea we had is to use an absolutely positioned button that sits on top of the textarea. That button would trigger the expand and collapse.

**Reply by ghost** [reply]
Hi @pguptaDS @richbachman 

I have tried to create same UX Priyanka mentioned using Text component & Flex css. 
I mocked up it, purely to give presentation n idea how we can achieve these using paste components, we can adjust CSS properties later also

[https://codesandbox.io/s/certification-full-view-d6dt0](https://codesandbox.io/s/certification-full-view-d6dt0
)

Please consider this as workaround not a proper solution as I am not using TextArea component.


Screenshot for more idea
<img width="500" alt="Screenshot 2021-08-06 at 8 09 10 PM" src="https://user-images.githubusercontent.com/72342162/128529477-ff428272-1c45-425b-98d7-27749c55e0f5.png">
<img width="500" alt="Screenshot 2021-08-06 at 8 09 26 PM" src="https://user-images.githubusercontent.com/72342162/128529492-4d8e94ea-e6ea-43e6-a717-033bd8b68f72.png">

**Reply by richbachman** [reply]
Hi @Kajal-Joshi

Thanks for putting this together. Here's a few more thoughts from the team:

There are a couple things nice about using Textarea:

- You can focus into the component and `cmd+A` and `cmd+C` to get the certificate. `Cmd+A` makes it a lot more precise to get the whole certificate text — there's more room for error when highlighting with a cursor.
- It's clear that there's more text in the scroll area (because the last line is cut off). The Box+Text implementation could make it seem like there's only 3 lines of text, and the "View full certificate" button could get lost especially when this is just one piece of a full page.

If the ability to view the full certificate is the most important piece to the feature, the Box+Text implementation is fine, but we'd recommend adding a gradient or something (and some [ScreenReaderOnly](https://paste.twilio.design/components/screen-reader-only) text) to indicate that there's more text than just those 3 lines. 

Adding a "Copy" button if you go the Box+Text route will be a big help as well.

**Reply by richbachman** [reply]
Hi @pguptaDS. Quick clarifying question. Is the purpose of this to copy the certificate to the clipboard, or to just view the certificate?

**Reply by pguptaDS** [reply]
@richbachman The main purpose is to be able to view the certificate.

**Reply by richbachman** [reply]
@pguptaDS circling back here finally. Sorry for the delay. Since the main purpose is to be able to view the certificate, it'd be best to use a textarea like this example: https://codesandbox.io/s/cert-example-forked-16c0g?file=/src/index.tsx

We're throwing around the idea of having a variant of Textarea with an adjustable height. Once we land on a decision there, we'll let you know.

**Reply by pguptaDS** [reply]
Hi @richbachman! Can you please also help us understand WHY the link within the text area is not desirable?

**Reply by richbachman** [reply]
@pguptaDS semantically, there's no way to put a button inside of a textarea. You could absolutely position a button so it visually appears to be inside the textarea. 

Something like: https://codesandbox.io/s/cert-example-forked-16c0g?file=/src/index.tsx (not perfect, just showing the concept). That way the button and textarea are both focusable elements.

A note on element order also. Ideally the expand/collapse button would be before the content it controls/reveals.
